android-json-form-wizard/src/main/java/com/vijay/jsonwizard/fragments/JsonFormFragment.java

```diff
@@ -670,72 +672,4 @@ public void setOnFieldsInvalid(OnFieldsInvalid onFieldsInvalid) {
     public static NativeFormsProperties getNativeFormProperties() {
         return nativeFormProperties;
     }
-
-    /**
-     * Skips blank by relevance steps when next is clicked on the json wizard forms.
-     */
-    public void skipStepsOnNextPressed() {
-        JSONObject formStep = getStep(getArguments().getString(JsonFormConstants.STEPNAME));
-        String next = formStep.optString(JsonFormConstants.NEXT, "");
-        if (StringUtils.isNotEmpty(next)) {
-            checkIfStepIsBlank(formStep);
-            if (shouldSkipStep()) {
-                next();
-            }
-        }
-    }
-
-    /**
-     * Skips blank by relevance steps when previous is clicked on the json wizard forms.
-     */
-    public void skipStepOnPreviousPressed() {
-        int currentFormStepNumber = getFormStepNumber();
-        for (int i = currentFormStepNumber; i >= 1; i--) {
-            JSONObject formStep = getJsonApi().getmJSONObject().optJSONObject(JsonFormConstants.STEP + i);
-            if (formStep != null) {
-                checkIfStepIsBlank(formStep);
-                if (shouldSkipStep()) {
-                    getFragmentManager().popBackStack();
-                } else {
-                    break;
-                }
-            }
-        }
-    }
-
-
-    /**
-     * Checks if a given step is blank due to relevance hidding all the widgets
-     *
-     * @param formStep {@link JSONObject}
-     */
-    private void checkIfStepIsBlank(JSONObject formStep) {
-        try {
-            if (formStep.has(JsonFormConstants.FIELDS)) {
-                JSONArray fields = formStep.getJSONArray(JsonFormConstants.FIELDS);
-                for (int i = 0; i < fields.length(); i++) {
-                    JSONObject field = fields.getJSONObject(i);
-                    if (field.has(JsonFormConstants.TYPE) && !JsonFormConstants.HIDDEN.equals(field.getString(JsonFormConstants.TYPE))) {
-                        boolean isVisible = field.optBoolean(JsonFormConstants.IS_VISIBLE, true);
-                        if (isVisible) {
-                            setShouldSkipStep(false);
-                            break;
-                        }
-                    }
-                }
-            }
-        } catch (JSONException e) {
-            Timber.e(e, "%s --> checkIfStepIsBlank", this.getClass().getCanonicalName());
-        }
-    }
-
-    /**
-     * Returns the current form step number which is used to figure out which steps to pop
-     * when previous is clicked.
-     *
-     * @return formNumber {@link Integer}
-     */
-    private int getFormStepNumber() {
-        return Integer.parseInt(getArguments().getString(JsonFormConstants.STEPNAME).substring(4));
-    }
```

Comment:
Why remove this here?

Reply:
The implementation and logic changed  .... has it broken you code?

Reply:
https://github.com/OpenSRP/opensrp-client-native-form/pull/407#discussion_r489401287

Reply:
Consider using `JsonWizardFormFragment`